source/slang/syntax.cpp

```diff
@@ -505,7 +505,7 @@ void Type::accept(IValVisitor* visitor, void* extra)
                         if (aggTypeDeclRef.getDecl()->memberDictionary.TryGetValue(assocTypeDecl->getName(), targetType))
                         {
                             if (auto typeDefDecl = dynamic_cast<TypeDefDecl*>(targetType))
-                                return typeDefDecl->type.type;
+                                return typeDefDecl->type.type->Substitute(aggTypeDeclRef.substitutions);
                             else
```

Comment:
It seems like this could also be handled by forming a `DeclRef` to `targetType` and then doing the dispatch on that, so you can do `GetType(typeDefDeclRef)` on a `DeclRef<TypeDefDecl>`.

I’m not sure that is a net improvement, but it might help if there are many cases to always start with a `DeclRef`.